### PR TITLE
Minor: Added k8s job

### LINKED ISSUE #35: Add verification k8s job to verify if app in created pod is up, running and reachable

### PR DESCRIPTION
**Pull request by Oleksandr6676**
**IMPORTANT NOTICE:** Merge after #56 is merged due to branch dependency!!!

Implemented k8s job to verify if app is up and running
Closes #35

### CODE REVIEW

**Comment by Oleksandr6676** [comment]
Hi Jan, hope you're doing great! I've implemented the first version of k8s job for busybox app, basically this job runs some commands inside the container and exit after completion. Also the task states that we should check reachability of the pod. How can we check this inside the k8s job? I tried to find some info in google but with no success.

**Review by JanMate — CHANGES_REQUESTED** [review]
The key idea is to have a k8s job that is based on a linux image providing `curl` command and sends request(s) to verify if another given pod/container is up and running. In other words, we are talking about 2 separate containers since one of them knows about existence of the other and the other have no clue. Even, it need not.

So please, rework.

**Comment by Oleksandr6676** [comment]
I implemented new version of k8s job, marked it as "work in progress" because it doesn't work with our core-app service for some reason, trying to fix it

File: kubernetes/core-pod.yml [hunk 1]
```diff
@@ -1,30 +1,30 @@
...
   containers:
-  - name: busybox-container
-    image: busybox
+  - name: core-app-container
+    image: jmatejka/happy-bank-core:1.0.0
     args:
     - /bin/sh
     - -c
-    - touch tmp/healthy; sleep infinity;
+    - touch healthy; ls; sleep infinity;
```
Comment (JanMate) [inline kubernetes/core-pod.yml:14]: Is this kind of health check needed when core app exposes health REST api endpoint?

File: kubernetes/core-job.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  backoffLimit: 2
+  template:
+    spec:
+      containers:
+        - name: curl-container
+          image: appropriate/curl
+          args:
+            - /bin/sh
+            - -c
+            - curl -v http://core-app-svc:80; sleep 20;
```
Comment (JanMate) [inline kubernetes/core-job.yml:16]: What is the sleep for?

**Review by JanMate — CHANGES_REQUESTED** [review]
@Oleksandr6676, check the pod logs, maybe you will find out what the issue is. And please, fix the issue in #56 at first. Then, fix this PR.

File: kubernetes/core-pod.yml (hunk 1 above)
Comment (JanMate) [reply kubernetes/core-pod.yml:14]: This comment will be solved in #56. So closing it.

**Review by JanMate — CHANGES_REQUESTED** [review]
Please, solve only the k8s job in this PR and issue.

File: kubernetes/core-job.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  backoffLimit: 2
+  template:
+    spec:
+      containers:
+        - name: curl-container
+          image: appropriate/curl
+          args:
+            - /bin/sh
+            - -c
+            - curl -v http://core-app-svc:5000/health;
```
Comment (JanMate) [inline kubernetes/core-job.yml:16]: There can be one optimization. If you check the image [Dockerfile](https://github.com/appropriate/docker-curl/blob/master/latest/Dockerfile), you will see the command is already set and there is only need to specify arguments.

```suggestion
            - -v
            - http://core-app-svc:5000/health
```

**Review by JanMate — CHANGES_REQUESTED** [review]
After refactor, it will be ready to merge.

File: kubernetes/core-job.yml (hunk 3 above)
Comment (Oleksandr6676) [reply kubernetes/core-job.yml:16]: Cool! It works, thank you for the hint

**Review by JanMate — APPROVED** [review]

**Pull request merged**